crates/configuration/src/configuration.rs

```diff
@@ -2,33 +2,36 @@
 
 use std::path::Path;
 
-use schemars::JsonSchema;
-use serde::{Deserialize, Serialize};
-
 use query_engine_metadata::metadata;
 
 use crate::environment::Environment;
-use crate::error::Error;
+use crate::error::{
+    MakeRuntimeConfigurationError, MultiError, ParseConfigurationError,
+    WriteParsedConfigurationError,
+};
 use crate::values::{IsolationLevel, PoolSettings};
 use crate::version3;
 use crate::version4;
 
-pub const CONFIGURATION_FILENAME: &str = "configuration.json";
-pub const CONFIGURATION_JSONSCHEMA_FILENAME: &str = "schema.json";
-
-/// The parsed connector configuration.
-#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize, JsonSchema)]
-#[serde(tag = "version")]
-pub enum RawConfiguration {
-    #[serde(rename = "3")]
+/// The parsed connector configuration. This data type is an enum with cases for each supported
+/// version.
+///
+/// It supports various uses:
+///
+/// * It can be turned into a `Configuration`, to be used at runtime.
+/// * It retains all information necessary to produce an equivalent serialized representation.
+/// * It supports updates between versions which may require more detailed information than is
+///   available in a `Configuration` (such as whether a native query was defined inline or in a
+///   file)
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum ParsedConfiguration {
```

Comment:
`RawConfiguration` is now called `ParsedConfiguration`, and it is no longer generically serializable.

Instead it defers to the implementations of different versions for ways to parse/write/maniuplate configuration value instances.